crates/miden-tx/src/errors/mod.rs

```diff
@@ -59,6 +60,10 @@ pub enum TransactionExecutorError {
     // case, the diagnostic is lost if the execution error is not explicitly unwrapped.
     #[error("failed to execute transaction kernel program:\n{}", PrintDiagnostic::new(.0))]
     TransactionProgramExecutionFailed(ExecutionError),
+    /// This variant can be matched on to get the effects of a transaction for signing purposes.
+    // It is boxed to avoid triggering clippy::result_large_err for functions that return this type.
+    #[error("transaction aborted with effects {0:?}")]
+    AbortWithTxEffects(Box<TransactionEffects>),
```

Comment:
Similar to one of the other comments, I would make this authorization specific.

My personal preference would probably be something like `Unauthorized`, but something like `AbortWithSigningMessage` or `AbortWithFingerprint` or `AbortUnauthorized` is probably fine too.